src/Credentials/ExternalAccountCredentials.php

```diff
@@ -122,6 +129,8 @@ public function __construct(
                 'workforce_pool_user_project should not be set for non-workforce pool credentials.'
             );
         }
+
+        $this->jsonKey = $jsonKey;
```

Comment:
I was thinking we should add a `getCacheKey` to the Credential Source classes (see `src/CredentialSource`), and use that instead. Then the credentials source classes can determine how to ensure their keys are "unique":

Something like:

```php
    public function getCacheKey()
    {
        return $this->auth->getSubjectTokenFetcher()->getCacheKey();
    }
```

There could be more to it as well, but this would be better than flattening the array IMHO